simpeg_drivers/components/data.py

```diff
@@ -171,14 +171,14 @@ def drape_locations(self, locations: np.ndarray) -> np.ndarray:
 
         return np.c_[distance_interp, locations[:, 2:]]
 
-    def filter(self, a):
+    def filter(self, obj: dict[str, np.ndarray] | np.ndarray, mask=None):
         """Remove vertices based on mask property."""
         if self.indices is None:
-            self.indices = np.where(self.mask)[0]
+            self.indices = np.where(mask)[0]
```

Comment:
That doesn't work.

Reply:
ok, @benk-mira , can you take it from here?
this change is coming from https://github.com/MiraGeoscience/simpeg-drivers/pull/23

Reply:
ok, brought in changes from https://github.com/MiraGeoscience/simpeg-drivers/pull/19 ( GEOPY-1477 )